launch_testing/launch_testing/__init__.py

```diff
@@ -83,7 +84,10 @@ def on_stdout_lines(self, lines):
             # We matched and we're in charge; shut myself down
             for td in self.launch_descriptor.task_descriptors:
                 if td.name == self.name:
-                    td.terminate()
```

Comment:
We could consider changing the behaviour of `terminate` itself, instead - I think `exit_on_match` might be the only thing that uses it at the moment